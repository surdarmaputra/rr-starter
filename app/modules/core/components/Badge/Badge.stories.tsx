import type { Meta, StoryObj } from '@storybook/react';

import { Badge, type BadgeProps } from './Badge';

type Story = StoryObj<typeof Badge>;

const meta: Meta = {
  component: Badge,
  title: 'Core/Components/Badge',
};

export default meta;

export const Default: Story = {
  args: {
    children: 'Label',
  },
  render: (args: BadgeProps) => <Badge {...args} />,
};
