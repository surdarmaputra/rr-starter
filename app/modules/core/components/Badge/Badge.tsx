import { cva, type VariantProps } from 'class-variance-authority';
import * as React from 'react';

import { cn } from '~/modules/core/libs/shadcn/utils';

const badgeVariants = cva(
  'inline-flex items-center rounded-full border border-divider px-2.5 py-0.5 text-xs font-bold transition-colors focus:outline-none focus:ring-2 focus:ring-secondary focus:ring-offset-2',
  {
    variants: {
      variant: {
        default: 'border-transparent bg-backdrop text-white',
        'default-outlined': 'border-letter-title text-letter-title',
        danger: 'border-transparent bg-danger text-white',
        'danger-outlined': 'border-danger text-danger',
        success: 'border-transparent bg-success text-white',
        'success-outlined': 'border-success text-success',
        warning: 'border-transparent bg-warning text-white',
        'warning-outlined': 'border-warning text-warning',
        primary: 'border-transparent bg-primary text-white',
        'primary-outlined': 'border-primary text-primary',
        secondary: 'border-transparent bg-secondary text-white',
        'secondary-outlined': 'border-secondary text-secondary',
      },
    },
    defaultVariants: {
      variant: 'default',
    },
  },
);

export interface BadgeProps
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof badgeVariants> {}

function Badge({ className, variant, ...props }: BadgeProps) {
  return (
    <div className={cn(badgeVariants({ variant }), className)} {...props} />
  );
}

export { Badge, badgeVariants };
