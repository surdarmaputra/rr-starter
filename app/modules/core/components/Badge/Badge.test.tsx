import { render, screen, within } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import { Badge } from './Badge';

describe('Badge', () => {
  function renderComponent() {
    return render(<Badge data-testid="badge-testid">Label</Badge>);
  }

  test('renders with data-testid', async () => {
    renderComponent();
    const badge = await screen.findByTestId('badge-testid');
    await within(badge).findByText(/Label/);
  });

  test('renders snapshot', async () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
