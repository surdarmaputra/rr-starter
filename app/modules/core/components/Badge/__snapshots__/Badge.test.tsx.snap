// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > renders snapshot 1`] = `
<div>
  <div
    class="inline-flex items-center rounded-full border px-4 py-0.5 text-xs font-bold transition-colors focus:outline-none focus:ring-2 focus:ring-secondary focus:ring-offset-2 border-transparent bg-backdrop text-white"
    data-testid="badge-testid"
  >
    Label
  </div>
</div>
`;
